src/factory-in-proc.js

```diff
@@ -4,84 +4,103 @@ const tmpDir = require('./utils/tmp-dir')
 const InProc = require('./ipfsd-in-proc')
 const merge = require('merge-options')
 const defaultConfig = require('./defaults/config.json')
+const { defaultRepo } = require('./utils/repo')
 
-/** @ignore @typedef {import("./index").SpawnOptions} SpawnOptions */
+/** @ignore @typedef {import("./index").IpfsOptions} IpfsOptions */
+/** @ignore @typedef {import("./index").FactoryOptions} FactoryOptions */
 
 /**
  * Factory to spawn in-proc JS-IPFS instances (aka in process nodes)
- * @class
- * @param {Object} options
- * @param {String} [options.type='proc'] - one of 'go', 'js' or 'proc', in this case this needs to be 'proc'
- * @param {String} [options.exec] - the path of the daemon executable or IPFS class in the case of `proc`
  */
 class FactoryInProc {
+  /**
+   * @param {FactoryOptions} options
+   */
   constructor (options) {
-    options = options || {}
-    if (options.type !== 'proc') {
-      throw new Error('This Factory only knows how to create in proc nodes')
-    }
-    this.options = options
+    /** @type FactoryOptions */
+    this.options = merge({
+      host: 'localhost',
+      port: 43134,
+      secure: false,
+      type: 'js',
+      defaultAddrs: false,
+      disposable: true,
+      env: process.env,
+      args: [],
+      ipfsHttp: {
+        path: require.resolve('ipfs-http-client'),
+        ref: require('ipfs-http-client')
+      },
+      ipfsApi: {
+        path: require.resolve('ipfs'),
+        ref: require('ipfs')
+      }
+      // ipfsBin: findBin(options.type || 'js')
+    }, options)
   }
 
   /**
    * Utility method to get a temporary directory
    * useful in browsers to be able to generate temp
    * repos manually
    *
-   * *Here for completeness*
-   *
    * @returns {Promise}
    */
   tmpDir () {
-    return Promise.resolve(tmpDir(true))
+    return Promise.resolve(tmpDir(this.options.type))
   }
 
   /**
    * Get the version of the currently used go-ipfs binary.
    *
-   * @param {Object} [options={}]
    * @returns {Promise}
    */
-  async version (options = {}) {
-    const node = new InProc(options)
+  async version () {
+    const node = new InProc(merge({
+      ipfsOptions: {
+        start: true,
+        init: true,
+        config: defaultConfig,
+        repo: tmpDir(this.options.type)
+      }
+    }, this.options))
     const v = await node.version()
```

Comment:
Shouldn't this new node be stopped? It's not used outside this function.

Reply:
yes this PR is still a WIP this is fixed already locally.
but im actually thinking about removing this method completely, not seeing much value in it plus we need to start the node to get the version, what do you think?